mirgecom/limiter.py

```diff
@@ -0,0 +1,98 @@
+""":mod:`mirgecom.limiter` is for limiters and limiter-related constructs.
+"""
+
+__copyright__ = """
+Copyright (C) 2022 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+from arraycontext import map_array_container
+
+from meshmode.dof_array import DOFArray
+
+from grudge.discretization import DiscretizationCollection
+
+from mirgecom.fluid import ConservedVars
+
+import grudge.op as op
+
+
+def limiter_liu_osher(dcoll: DiscretizationCollection, state):    
```

Comment:
Take in two `dd`s, and two `state`s:
- One on which the nodal minima/maxima are taken.
- One that actually has the limiting applied.